Provide memfd_create() and its flags without the Linux option.

The function sits on a sysdep any port can implement, but it and the MFD_*
flags are compiled and declared only under the Linux option -- which needs
Linux's kernel headers. Sirius has memfd_create and no Linux headers. With
this, _GNU_SOURCE alone is enough.

--- a/abis/linux/vm-flags.h
+++ b/abis/linux/vm-flags.h
@@ -75,11 +75,14 @@
 #define MREMAP_MAYMOVE 1
 #define MREMAP_FIXED 2
 
+#endif /* defined(_GNU_SOURCE) */
+
+#endif /* __MLIBC_LINUX_OPTION */
+
+#if defined(_GNU_SOURCE)
 #define MFD_CLOEXEC 1U
 #define MFD_ALLOW_SEALING 2U
 #define MFD_HUGETLB 4U
 #endif /* defined(_GNU_SOURCE) */
 
-#endif /* __MLIBC_LINUX_OPTION */
-
 #endif /* _ABIBITS_VM_FLAGS_H */
--- a/options/posix/generic/sys-mman.cpp
+++ b/options/posix/generic/sys-mman.cpp
@@ -139,17 +139,6 @@
 	__builtin_unreachable();
 }
 
-int memfd_create(const char *name, unsigned int flags) {
-	int ret = -1;
-
-	if(int e = mlibc::sysdep_or_enosys<MemfdCreate>(name, flags, &ret)) {
-		errno = e;
-		return -1;
-	}
-
-	return ret;
-}
-
 int madvise(void *addr, size_t length, int advice) {
 	if(int e = mlibc::sysdep_or_enosys<Madvise>(addr, length, advice)) {
 		errno = e;
@@ -167,3 +156,14 @@
 	return 0;
 }
 #endif /* __MLIBC_LINUX_OPTION */
+
+int memfd_create(const char *name, unsigned int flags) {
+	int ret = -1;
+
+	if(int e = mlibc::sysdep_or_enosys<MemfdCreate>(name, flags, &ret)) {
+		errno = e;
+		return -1;
+	}
+
+	return ret;
+}
--- a/options/posix/include/sys/mman.h
+++ b/options/posix/include/sys/mman.h
@@ -35,9 +35,12 @@
 #if __MLIBC_LINUX_OPTION && defined(_GNU_SOURCE)
 void *mremap(void *__old_address, size_t __old_size, size_t __new_size, int __flags, ...);
 int remap_file_pages(void *__addr, size_t __size, int __prot, size_t __pgoff, int __flags);
-int memfd_create(const char *__name, unsigned int __flags);
 #endif /* __MLIBC_LINUX_OPTION && defined(_GNU_SOURCE) */
 
+#if defined(_GNU_SOURCE)
+int memfd_create(const char *__name, unsigned int __flags);
+#endif /* defined(_GNU_SOURCE) */
+
 #if __MLIBC_LINUX_OPTION && defined(_DEFAULT_SOURCE)
 int madvise(void *__addr, size_t __size, int __advise);
 int mincore(void *__addr, size_t __size, unsigned char *__vec);
